float32 joint1
float32 joint2